### PR TITLE
Use upstream openWakeWord

### PR DESCRIPTION
**Pull request by synesthesiam**
Switches to using the upstream [openWakeWord](https://github.com/dscripka/openWakeWord) Python library instead of a custom fork.

Some things not yet implemented:
* Custom verifier models
* Silero VAD

### CODE REVIEW

**Comment by qJake** [comment]
I'm new to this repository, and I wanted to test this branch out because `master` was giving me issues.

I checked it out, installed the requirements, and it is running successfully... but I had to remove the `--preload-model` parameter. I use a custom wake word model that works in `master` (by specifying `--custom-model-dir` and `--preload-model`), but I couldn't get that to work on this branch.

I guessed that maybe OWW would be configured via Home Assistant then (since you can change the wake word there), but when I went to add it to the Wyoming Protocol like this:

image

But it fails:

image

I also tried "ok nabu" and "hey jarvis", figuring maybe it was hard-coded for testing... and nope. Didn't work either. 🙃 

What am I missing here? I'd love to help test this branch!

**Comment by synesthesiam** [comment]
Thanks for trying out the branch! Do you see any errors in the logs when the connection fails? I'll double check, but it was working for me.

I removed the preload model argument since it isn't needed with OWW proper.

**Comment by qJake** [comment]
> Thanks for trying out the branch! Do you see any errors in the logs when the connection fails? I'll double check, but it was working for me.

I didn't see anything, but let me try again with --debug and I'll report back.

For what it's worth, on the same device, the Satellite can connect just fine on `:10700`.

**Comment by Graimalkin** [comment]
If you restart OpenWakeWord with Satellites already connected to it, handle_event crashes because self.model is None.

You need to kill the satellites **THEN** restart OpenWakeWord, so that it doesn't get a AudioChunk type message before it can init the model. This is obnoxious when you're running multiple remote satellites that aren't on the same machine as the OpenWakeWord container. 😛 

Screenshot shows satellites up and container restarting ( and erroring ). Then satellites off and OpenWakeWord container restarting successfully and allowing the client to connect.

You'll probably want to put some sort of self.model init function in for if you get an AudioChunk type message and self.model is None

image

**Comment by synesthesiam** [comment]
Thanks @Graimalkin this is easy to fix 👍

**Comment by canberkoguz** [comment]
hi @synesthesiam, thanks for the great work! 
as a developer, is there anything that I can help to get custom verifier model support?

**Comment by qJake** [comment]
I was able to finally test this, and while it does show up in Home Assistant correctly, I'm getting this error on startup:

```
Jul 14 21:49:49 voice systemd[1]: Started wyoming-openwakeword.service - Wyoming OpenWakeWord.
Jul 14 21:49:52 voice run[10677]: INFO:root:Ready
Jul 14 21:49:53 voice run[10677]: ERROR:asyncio:Task exception was never retrieved
Jul 14 21:49:53 voice run[10677]: future: <Task finished name='wyoming event handler' coro=<AsyncEventHandler.run() done, defined at /home/voice/wyoming-openwakeword/.venv/lib/python3.11/site-packages/wyoming/server.py:31> exception=AssertionError()>
Jul 14 21:49:53 voice run[10677]: Traceback (most recent call last):
Jul 14 21:49:53 voice run[10677]:   File "/home/voice/wyoming-openwakeword/.venv/lib/python3.11/site-packages/wyoming/server.py", line 41, in run
Jul 14 21:49:53 voice run[10677]:     if not (await self.handle_event(event)):
Jul 14 21:49:53 voice run[10677]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 14 21:49:53 voice run[10677]:   File "/home/voice/wyoming-openwakeword/wyoming_openwakeword/handler.py", line 124, in handle_event
Jul 14 21:49:53 voice run[10677]:     assert self.model is not None
Jul 14 21:49:53 voice run[10677]:            ^^^^^^^^^^^^^^^^^^^^^^
Jul 14 21:49:53 voice run[10677]: AssertionError

```

Any ideas? I'm no good with Python async. 😁

**Comment by matt1432** [comment]
I've been running this PR for a while now and it works well.

However, I can't get the VAD to work. I always get this error:

```console
Dec 18 21:37:40 homie wyoming-openwakeword[354739]: DEBUG:root:Namespace(models_dir=PosixPath('/nix/store/3sgs163lbwd7ghzpjkl03z5cj2ida5vf-wyoming-openwakeword-1.10.0-unstable/lib/python3.11/site-packages/wyoming_openwakeword/models'), uri='tcp://127.0.0.1:10400', custom_model_dir=['/nix/store/asbacij2gzpr0jadj0rvaadq6gffxacb-source/en/yo_homie'], threshold=0.55, output_dir=None, refractory_seconds=0.5, vad_threshold=0.5, debug=True, log_format='%(levelname)s:%(name)s:%(message)s', debug_probability=False, version=False)
Dec 18 21:37:40 homie wyoming-openwakeword[354739]: DEBUG:root:Using Silero VAD (threshold=0.5)
Dec 18 21:37:40 homie wyoming-openwakeword[354739]: INFO:root:Ready
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: DEBUG:wyoming_openwakeword.handler:Client connected: 99076460688204
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: DEBUG:wyoming_openwakeword.handler:Loading openWakeWord models: ['/nix/store/asbacij2gzpr0jadj0rvaadq6gffxacb-source/en/yo_homie/yo_homie.tflite']
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: DEBUG:wyoming_openwakeword.handler:Receiving audio from client: 99076460688204
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: 2024-12-18 21:38:46.553334408 [E:onnxruntime:, sequential_executor.cc:516 ExecuteKernel] Non-zero status code returned while running Conv node. Name:'Conv_132' Status Message: Invalid input shape: {4}
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: 2024-12-18 21:38:46.553359767 [E:onnxruntime:, sequential_executor.cc:516 ExecuteKernel] Non-zero status code returned while running If node. Name:'If_25' Status Message: Non-zero status code returned while running Conv node. Name:'Conv_132' Status Message: Invalid input shape: {4}
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: DEBUG:wyoming_openwakeword.handler:Client disconnected: 99076460688204
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: ERROR:asyncio:Task exception was never retrieved
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: future: <Task finished name='wyoming event handler' coro=<AsyncEventHandler.run() done, defined at /nix/store/jir30wdrp85nff14idbw4n9p4zb713vv-python3.11-wyoming-1.5.4/lib/python3.11/site-packages/wyoming/server.py:31> exception=InvalidArgument("[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running If node. Name:'If_25' Status Message: Non-zero status code returned while running Conv node. Name:'Conv_132' Status Message: Invalid input shape: {4}")>
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: Traceback (most recent call last):
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/jir30wdrp85nff14idbw4n9p4zb713vv-python3.11-wyoming-1.5.4/lib/python3.11/site-packages/wyoming/server.py", line 41, in run
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     if not (await self.handle_event(event)):
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/3sgs163lbwd7ghzpjkl03z5cj2ida5vf-wyoming-openwakeword-1.10.0-unstable/lib/python3.11/site-packages/wyoming_openwakeword/handler.py", line 133, in handle_event
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     self.model.predict(chunk_array)
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/rpwz5r4y0bqs6zpwjcvpp11y0lg9kara-openwakeword-0.6.0/lib/python3.11/site-packages/openwakeword/model.py", line 370, in predict
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     self.vad(x)
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/rpwz5r4y0bqs6zpwjcvpp11y0lg9kara-openwakeword-0.6.0/lib/python3.11/site-packages/openwakeword/vad.py", line 130, in __call__
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     self.prediction_buffer.append(self.predict(x, frame_size))
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/rpwz5r4y0bqs6zpwjcvpp11y0lg9kara-openwakeword-0.6.0/lib/python3.11/site-packages/openwakeword/vad.py", line 123, in predict
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     ort_outs = self.model.run(None, ort_inputs)
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:   File "/nix/store/axm3krv1lrhz3n794p11qli7a44xs9dx-python3.11-onnxruntime-1.18.1/lib/python3.11/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 220, in run
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:     return self._sess.run(output_names, input_feed, run_options)
Dec 18 21:38:46 homie wyoming-openwakeword[354739]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 18 21:38:46 homie wyoming-openwakeword[354739]: onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running If node. Name:'If_25' Status Message: Non-zero status code returned while running Conv node. Name:'Conv_132' Status Message: Invalid input shape: {4}
```

**Comment by AnkushMalaker** [comment]
> I was able to finally test this, and while it does show up in Home Assistant correctly, I'm getting this error on startup:
> 
> ```
> Jul 14 21:49:49 voice systemd[1]: Started wyoming-openwakeword.service - Wyoming OpenWakeWord.
> Jul 14 21:49:52 voice run[10677]: INFO:root:Ready
> Jul 14 21:49:53 voice run[10677]: ERROR:asyncio:Task exception was never retrieved
> Jul 14 21:49:53 voice run[10677]: future: <Task finished name='wyoming event handler' coro=<AsyncEventHandler.run() done, defined at /home/voice/wyoming-openwakeword/.venv/lib/python3.11/site-packages/wyoming/server.py:31> exception=AssertionError()>
> Jul 14 21:49:53 voice run[10677]: Traceback (most recent call last):
> Jul 14 21:49:53 voice run[10677]:   File "/home/voice/wyoming-openwakeword/.venv/lib/python3.11/site-packages/wyoming/server.py", line 41, in run
> Jul 14 21:49:53 voice run[10677]:     if not (await self.handle_event(event)):
> Jul 14 21:49:53 voice run[10677]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Jul 14 21:49:53 voice run[10677]:   File "/home/voice/wyoming-openwakeword/wyoming_openwakeword/handler.py", line 124, in handle_event
> Jul 14 21:49:53 voice run[10677]:     assert self.model is not None
> Jul 14 21:49:53 voice run[10677]:            ^^^^^^^^^^^^^^^^^^^^^^
> Jul 14 21:49:53 voice run[10677]: AssertionError
> ```
> 
> Any ideas? I'm no good with Python async. 😁

I get the same. @matt1432 , since you say this has been working well, can you share setup steps so I can quickly find what I may be doing differently to cause this?

**Comment by edurenye** [comment]
Seems like [openWakeWord](https://github.com/dscripka/openWakeWord) is fine with NumPy 2, and NumPy 2 is compatible with builds using NumPy 1. So, do we still need `numpy<2` to be set in https://github.com/rhasspy/wyoming-openwakeword/blob/master/requirements.txt for this repo?
